src/iam.ts

```diff
@@ -98,6 +98,21 @@ export class IAM extends IAMBase {
 
     return { isMetamaskPresent: !!provider, chainId };
   }
+
+  static isSessionActive() {
+    if (window && window.sessionStorage) {
+      return Boolean(sessionStorage.getItem(PUBLIC_KEY));
+    }
+    return false;
+  }
+
+  static getProviderType() {
+    if (window && window.sessionStorage) {
+      const providerType = sessionStorage.getItem(WALLET_PROVIDER) as WalletProvider | null;
+      return providerType;
+    }
+    return null;
+  }
```

Comment:
Just curious, why are are these functions static? (maybe there is a good reason that I just can't see 😄 ) I'm just thinking that if they were instance members, then they could use the `_providerType` and `_publicKey` properties which would mean that the use of sessionStorage would be a little more encapsulated?

Reply:
Yeah good point, I'll do that 👍